pageserver/src/tenant.rs

```diff
@@ -1226,8 +1226,20 @@ impl Tenant {
             "Cannot create timelines on inactive tenant"
         );
 
-        if self.get_timeline(new_timeline_id, false).is_ok() {
+        if let Ok(existing) = self.get_timeline(new_timeline_id, false) {
             debug!("timeline {new_timeline_id} already exists");
+
+            if let Some(remote_client) = existing.remote_client.as_ref() {
+                // wait for uploads to complete even if this is a random old/very active timeline
+                // because if we get shutdown while waiting for the uploads on creation path (later
+                // in this method), the request will be retried and during startup we already
+                // scheduled the uploads so they have either completed or are completing right now.
```

Comment:
```suggestion
                // Wait for uploads to complete, so that when we return Ok, the timeline is known to
                // be durable on remote storage. Just like we do at the end of this function, after we have
                // created the timeline ourselves.
                //
                // We only really care that the initial version of `index_part.json` has been uploaded. That's enough to remember that
                // the timeline exists. However, there is no function to wait specifically for that
                // so we just wait for all in-progress uploads to finish.
```